### ISSUE TITLE
Errors importing Arrow Core 1.0.0 dependency in MPP

### ISSUE THREAD

**Issue by Joe3112** [arrow-kt/arrow#2534]
When trying to add dependency on arrow core 1.0.0:

```
plugins {
    kotlin("multiplatform") version "1.5.31"
}

group = "com.middlepoint"
version = "1.0-SNAPSHOT"

repositories {
    mavenCentral()
}

kotlin {
    jvm {
        compilations.all {
            kotlinOptions.jvmTarget = "1.8"
        }
        testRuns["test"].executionTask.configure {
            useJUnitPlatform()
        }
    }
    js(LEGACY) {
        browser {
            commonWebpackConfig {
                cssSupport.enabled = true
            }
        }
    }
    val hostOs = System.getProperty("os.name")
    val isMingwX64 = hostOs.startsWith("Windows")
    val nativeTarget = when {
        hostOs == "Mac OS X" -> macosX64("native")
        hostOs == "Linux" -> linuxX64("native")
        isMingwX64 -> mingwX64("native")
        else -> throw GradleException("Host OS is not supported in Kotlin/Native.")
    }

    
    sourceSets {
        val commonMain by getting {
            dependencies {
                implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.5.2")
                implementation("io.arrow-kt:arrow-core:1.0.0")
            }
        }
        val commonTest by getting {
            dependencies {
                implementation(kotlin("test"))
            }
        }
        val jvmMain by getting
        val jvmTest by getting {
            dependencies {
                implementation( "org.mockito.kotlin:mockito-kotlin:3.2.0")
                implementation( "org.jetbrains.kotlinx:kotlinx-coroutines-test:1.5.2")
            }
        }
        val jsMain by getting
        val jsTest by getting
        val nativeMain by getting
        val nativeTest by getting
    }
}
```

Getting the following error:

```
Failed building KotlinMPPGradleModel
org.gradle.internal.resolve.ArtifactNotFoundException: Could not find arrow-core-1.0.0-samplessources.jar (io.arrow-kt:arrow-core:1.0.0).
Searched in the following locations:
    https://repo.maven.apache.org/maven2/io/arrow-kt/arrow-core/1.0.0/arrow-core-1.0.0-samplessources.jar
```

When changing to use  new JS backend IR (`js(IR)`) the error message changes to:

```
Could not find io.arrow-kt:arrow-core-macosx64:unspecified.
Searched in the following locations:
  - https://repo.maven.apache.org/maven2/io/arrow-kt/arrow-core-macosx64/unspecified/arrow-core-macosx64-unspecified.pom
If the artifact you are trying to retrieve can be found in the repository but without metadata in 'Maven POM' format, you need to adjust the 'metadataSources { ... }' of the repository declaration.
Required by:
    project : > io.arrow-kt:arrow-core:1.0.0

Possible solution:
 - Declare repository providing the artifact, see the documentation at https://docs.gradle.org/current/userguide/declaring_repositories.html
```

**Comment by nomisRev** [comment]
Hey @Joe3112,

`JS(LEGACY)` is not published from Arrow. Is this something libraries should still be adding?

There seems to be something wrong with `1.0.0` metadata for `macos`, this should be solved if you attempt to resolve `1.0.1-SNAPSHOT`. We're resolving the issue with Apple targets asap. Windows & Linux should resolve correctly.

**Comment by Joe3112** [comment]
Hi @nomisRev, I thought about checking the snapshot but wanted to let you guys know about the issue! cheers and congrats on reaching V1 :)

**Comment by nomisRev** [comment]
Thank you very much @Joe3112! 🙏 This will be fixed very soon, at which we'll release 1.0.1 with Kotlin 1.5.31.
